spec/function-builder.spec.ts

```diff
@@ -150,38 +150,22 @@ describe('FunctionBuilder', () => {
     }).to.throw(Error, 'RuntimeOptions.timeoutSeconds');
   });
 
-  it('should throw an error if user chooses a failurePolicy of type number', () => {
+  it('should throw an error if user chooses a failurePolicy which is neither an object nor a boolean', () => {
```

Comment:
Should we have another test case where we cast `as object` for completeness?

Reply:
The cast is only to make compiler happy, it doesn't change anything at runtime - additional test seems unnecessary, WDYT?

Reply:
I see your point - I was mostly thinking it's nice to have the test documenting the usage as well, so that was my main objective. It seems a bit confusing in the test name to say it can be something other than boolean or object but then cast to boolean only. I'd probably like to still have an extra case for it.

Reply:
Ok, I see. I restructured this test a little so it's more clear what's tested here, it changes two things:
- Expected error message is more precise and contains `must be a boolean or an object`, so it correlates better with the name of the test.
- The cast I put in place now casts to `RuntimeOptions['failurePolicy']` instead of casting to `boolean`, so neither `boolean` nor `object` are favorized.

See ac09c189a28b5514f3b78d7eff7b32368e46a469

Reply:
This looks good, thanks.